frontend/ui/widget/datetimewidget.lua

```diff
@@ -158,6 +185,21 @@ function DateTimeWidget:createLayout()
             value_step = 1,
             value_hold_step = self.year_hold_step or 4,
             width = number_picker_widgets_width,
+            picker_updated_callback = function(value)
+                if self.month and self.day then
+                    local current_month = self.month_widget:getValue()
+                    local current_day = self.day_widget:getValue()
+                    -- check if we're on the 29th February
+                    if current_month == 2 and current_day == 29 then
+                        -- check if the new year-value is a leap year
+                        if value % 4 ~= 0 or (value % 100 == 0 and value % 400 ~= 0) then
+                            -- change to 28th February
```

Comment:
Shouldn't that also use the daysInMonth thing like on line 221? Looks good otherwise.